arangod/StorageEngine/VPackSortMigration.cpp

```diff
@@ -0,0 +1,294 @@
+                    LOG_TOPIC("42540", TRACE, Logger::ENGINES)
+                        << "VPackSortMigration: comparing keys: "
+                        << key.ToString(true) << " with "
+                        << rocksdb::Slice(prev_key.data(), prev_key.size())
+                               .ToString(true);
+                    // Note that there is an implicit conversion from
+                    // std::string to rocksdb::Slice:
+                    if (newComparator.Compare(prev_key, key) > 0) {
+                      LOG_TOPIC("42541", WARN, Logger::ENGINES)
+                          << "VPackSortMigration: found problematic key for "
+                             "database "
+                          << name << ", collection " << collection->name()
+                          << ", index with ID " << index->id().id()
+                          << " and name " << index->name();
+                      alarm = true;
+                      break;
+                    }
+                  }
+                  prev_key.assign(key.data(), key.size());
+                }
+                if (alarm) {
+                  problemFound = true;
+                  {
+                    VPackObjectBuilder guardO2(&result);
+                    guardO2->add("database", VPackValue(name));
+                    guardO2->add("collection", VPackValue(collection->name()));
+                    guardO2->add("indexId", VPackValue(index->id().id()));
+                    guardO2->add("indexName", VPackValue(index->name()));
+                  }
+                }
+              }
+            }
+          }
+        });
+      }
+    }
+
+    if (!problemFound) {
+      guardO->add(StaticStrings::Error, VPackValue(false));
+      guardO->add(StaticStrings::ErrorCode, VPackValue(0));
+      guardO->add(StaticStrings::ErrorMessage,
+                  VPackValue("all good with sorting order"));
+    } else {
+      guardO->add(StaticStrings::Error, VPackValue(true));
+      guardO->add(StaticStrings::ErrorCode,
+                  VPackValue(TRI_ERROR_ARANGO_INDEX_HAS_LEGACY_SORTED_KEYS));
+      guardO->add(StaticStrings::ErrorMessage,
+                  VPackValue("some indexes have legacy sorted keys"));
+    }
+  }
+
+  return {};
+}
+
+Result migrateVPackIndexSorting(TRI_vocbase_t& vocbase, VPackBuilder& result) {
+  // Just for the sake of completeness, restrict ourselves to the RocksDB
+  // storage engine:
+  result.clear();
+  auto& engineSelectorFeature =
+      vocbase.server().getFeature<EngineSelectorFeature>();
+  if (!engineSelectorFeature.isRocksDB()) {
+    {
+      VPackObjectBuilder guard(&result);
+      guard->add(StaticStrings::Error, VPackValue(false));
+      guard->add(StaticStrings::ErrorCode, VPackValue(0));
+      guard->add(
+          StaticStrings::ErrorMessage,
+          VPackValue("VPack sorting migration is unnecessary for storage "
+                     "engines other than RocksDB"));
+    }
+    return {};
```

Comment:
should this return an error here?

Reply:
My thinking was that the API call was successfull, albeit unneeded. Happy to discuss.